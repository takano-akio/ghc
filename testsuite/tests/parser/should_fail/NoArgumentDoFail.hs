module NoArgumentDoFail where

when :: Bool -> IO () -> IO ()
when True a = a
when False _ = return ()

foo :: IO ()
foo = when True do
  return ()
