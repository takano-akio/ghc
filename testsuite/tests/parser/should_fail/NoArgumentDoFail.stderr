
NoArgumentDoFail.hs:8:17: error:
    Unexpected do block in function application:
        do { return () }
    You could write it with parentheses
    Or perhaps you meant to enable ArgumentDo?
